File: Ticketing_Portal_Project/Backend/Libraries/TicketingLibrary/Models/SLA.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TicketingLibrary.Models
{
    [Table("SLA")]
    public class SLA
    {
        [Key]
        [Column(TypeName = "CHAR(4)")]
        public string? SLAId { get; set; }

        [Column(TypeName = "VARCHAR(50)")]
        public string? SLAName { get; set; }

        [Column(TypeName = "VARCHAR(20)")]
        public string? Priority { get; set; }
        
        public int ResponseTime { get; set; }
        public int ResolutionHours { get; set; }

        public virtual ICollection<TicketType> TicketTypes { get; set; } = new List<TicketType>();
    }
}

File: Ticketing_Portal_Project/Frontend/angular-app/src/app/models/SLA.ts
export class SLA {
  public slaId: string;
  public slaName: string;
  public priority: string;
  public responseTime: number;
  public resolutionHours: number;

  constructor(
    slaId: string,
    slaName: string,
    priority: string,
    responseTime: number,
    resolutionHours: number
  ) {
    this.slaId = slaId;
    this.slaName = slaName;
    this.priority = priority;
    this.responseTime = responseTime;
    this.resolutionHours = resolutionHours;
  }
}
